### ISSUE TITLE
setting of inline amplifier (ILA)

### ISSUE THREAD

**Issue by ojnas** [OpenROADM/OpenROADM_MSA_Public#40]
According to the OpenROADM device YANG model (org-openroadm-device.yang), both "target-gain" and "egress-average-channel-power" should be provided by the controller to an ILA for operating it in gainLoss control-mode. Why is it not sufficient to provide target-gain? How is an ILA supposed to use the channel-power information anyway when it only measures total aggregate power.

**Comment by Fredrik-Smartoptics** [comment]
I am also interested in this question linked to how the ILA is supposed to work. Will the SDN controller set the "target-gain" only then there is no need for an OCM in the device, the controller has done all the calculations. If the SDN controller only sets the "egress-average-channel-power" then as I see it an OCM is needed. Aggregate monitors are good and can work as a substitute when higher channel counts is used. But when lower channel counts are present the total Rx and Tx power cannot accurately be used for calculating channel power.

**Comment by orenais** [comment]
> 
> 
> According to the OpenROADM device YANG model (org-openroadm-device.yang), both "target-gain" and "egress-average-channel-power" should be provided by the controller to an ILA for operating it in gainLoss control-mode. Why is it not sufficient to provide target-gain? How is an ILA supposed to use the channel-power information anyway when it only measures total aggregate power.

Sorry for replying so late Jonas. Completely missed it. First, I am not an expert on this topic. I can only provide feedback on my understanding of some of the discussions I followed. From what I understood (and remember...), egress-average-channel power is supposed to be used to adjust a VOA in the optical amplifier, because for some conditions (span length/attenuation), we may want to use the amplifier in its optimal operational conditions which minimizes the noise figure (pump power and gain setting) which implies to use of VOA to reduce the power at the output (for short spans). The power is not supposed to be monitored on a channel basis. The value is provided considering a 50GHz channel width, considering we will work at constant power spectral density. Does it make sense to you?

**Comment by orenais** [comment]
> 
> 
> I am also interested in this question linked to how the ILA is supposed to work. Will the SDN controller set the "target-gain" only then there is no need for an OCM in the device, the controller has done all the calculations. If the SDN controller only sets the "egress-average-channel-power" then as I see it an OCM is needed. Aggregate monitors are good and can work as a substitute when higher channel counts is used. But when lower channel counts are present the total Rx and Tx power cannot accurately be used for calculating channel power.

Hi Fredrik. I guess the answer I provided to Jonas shall clarify some points. Both target gain and egress-average-channel-power are read/write parameters, provided by the controller. The last is used by the Device embedded controller to set the VOA for the reason explained above, the target gain is used to provide information on how much of the attenuation of preceeding span shall be compensated. From what I remember, I guess OEMs do not include OCM in the amplifiers they provide. (Pieces of the puzzle starts to assembly again in my mind, when I am digging further in the existing documents, some of them being a bit old (we modeled amplifiers in release 2 .... ;-))

**Comment by ojnas** [comment]
Hi Olivier, what you say about adjusting a VOA for optimal operation makes sense. However, the description of "target-gain" in the yang model says:

`Target overall Amplifier Signal gain, excluding ASE, including VOA attenuation.`

If the "target-gain" already includes the VOA then I still don't understand what extra information "egress-average-channel-power" provides.

**Comment by ojnas** [comment]
Yes, I understand all this. I just don't understand why the target-gain is not sufficient for the device to set its operating parameters (EDFA gain and VOA attenuation).
